float32 lspeed
float32 rspeed